src/storage/migrations.js

```diff
@@ -475,6 +475,28 @@ const migrationsInner: {| [string]: (LessPartialState) => LessPartialState |} =
   // Fix tiny case where an `undefined` could sneak in on ACCOUNT_SWITCH
   '57': state => ({ ...state, accounts: state.accounts.filter(Boolean) }),
 
+  // Remove malformed partial `Account`s. Perhaps we could salvage some, if
+  // they're just missing e.g. `zulipFeatureLevel`… but we don't think the
+  // bug affected many people, and when it did, it quickly followed a state
+  // where `accounts` was empty, probably because the user asked to remove
+  // the last account.
+  '58': state => ({
+    ...state,
+    accounts: state.accounts.filter(account =>
+      [
+        'realm',
+        'apiKey',
+        'email',
+        'userId',
+        'zulipVersion',
+        'zulipFeatureLevel',
+        'ackedPushToken',
+        'lastDismissedServerPushSetupNotice',
+        // eslint-disable-next-line no-prototype-builtins
+      ].every(key => account.hasOwnProperty(key)),
```

Comment:
This lint rule seems like it's giving good advice:
https://eslint.org/docs/latest/rules/no-prototype-builtins
> To avoid subtle bugs like this, it’s better to always call these methods from `Object.prototype`. For example, `foo.hasOwnProperty("bar")` should be replaced with `Object.prototype.hasOwnProperty.call(foo, "bar")`.

In newer JS there's also `Object.hasOwn` as a more concise equivalent to that latter form:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn
but that was new in iOS Safari 15.4, so we can't use it yet.

Reply:
Ah right, yeah. Thanks!